client/hotrod/src/main/java/org/infinispan/hotrod/impl/transport/netty/ByteBufUtil.java

```diff
@@ -23,9 +23,44 @@ public static byte[] readArray(ByteBuf buf) {
       return bytes;
    }
 
-   public static String readString(ByteBuf buf) {
-      byte[] strContent = readArray(buf);
-      return new String(strContent, HotRodConstants.HOTROD_STRING_CHARSET);
+   public static byte[] readMaybeArray(ByteBuf bf) {
+      int length = readMaybeVInt(bf);
+      if (length == Integer.MIN_VALUE) {
+         return null;
+      }
+      return readMaybeRangedArray(bf, length);
+   }
+
+   public static byte[] readMaybeRangedArray(ByteBuf bf, int length) {
+      if (!bf.isReadable(length)) {
+         bf.resetReaderIndex();
+         return null;
+      } else {
+         if (length == 0) {
```

Comment:
This if block here should be outside of the outer if I think. If the last values on the ByteBuf was a zero sized array it would never finish I believe.